src/Entity/ServerInterface.php

```diff
@@ -60,7 +60,7 @@ public function removeAllPersistedQueryInstances();
   /**
    * Returns the current persisted queries set.
    *
-   * @return \Drupal\graphql\Plugin\PersistedQueryPluginInterface[]
+   * @return \Drupal\graphql\Plugin\PersistedQueryPluginInterface[]|null
```

Comment:
This would technically be a breaking change. Since the following code would break:
```
foreach ($this->getPersistedQueryInstances() as $instance) {}
```

I suspect this is something suggested by PHPStan? I can see where it comes from [in the Server implementation](https://github.com/drupal-graphql/graphql/blob/8.x-4.x/src/Entity/Server.php#L435). That implementation uses PHP's "convert NULL to empty array on array access" side-effect.

I think the better (and more ergonomic for callers) fix would be to add `$this-> persisted_query_instances = [];`, right after the `is_null` if statement's closing bracked at the top of the function. That has a few positive effects:

1. The function never returns NULL so calling it is easier (because I'm only dealing with arrays)
2. If there are no plugins defined then we don't search for them every time `getPersistedQueryInstances` is called (which happens now, because in the no-plugin case, `$this->persisted_query_instances` is still `NULL` when the function is called after the first time).

For clarity we should probably just split that change out into a separate PR since it's not necessarily related to APQ? Then this PR doesn't need to be blocked by that change either :)